darwin/config.py

```diff
@@ -4,61 +4,107 @@
 
-        if isinstance(key, str):
-            key = key.split("/")
-        key, *keys = key
-        if len(keys) == 0:
-            return acc.get(key)
-        return self.get(keys, acc.get(key))
+        while True:
+            if isinstance(key, str):
+                key = key.split("/")
+            key, *keys = key
+            acc = acc.get(key)
+            if acc is None:
+                return default
+            if len(keys) == 0:
+                return acc
+            else:
+                key = keys
 
-    def parse(self):
-        """Parses the YAML configuration file"""
-        with open(self._path, "r") as stream:
-            return yaml.safe_load(stream)
-
-    def write(self, key: Union[str, List[str]], value: Any, save: bool = True):
+    def put(self, key: Union[str, List[str]], value: any, save: bool = True):
         """Sets value for specified key
         
         Args:
         - key: the key where the value is going to be stored.
         It can be formatted as a simple string, or as a path/like/string to fetch nested values.
         - value: the value to be set"""
-
         if isinstance(key, str):
             key = key.split("/")
+
+        pointer = self._data
+
         for k in key[:-1]:
-            self._data = self._data.setdefault(k, {})
-        self._data[key[-1]] = value
+            pointer = pointer.setdefault(k, {})
+        pointer[key[-1]] = value
+
         if save:
             self._save()
 
     def _save(self):
         """Persist the configuration to the file system"""
         with io.open(self._path, "w", encoding="utf8") as f:
             yaml.dump(self._data, f, default_flow_style=False, allow_unicode=True)
+
+    def set_team(self, team: str, api_key: str):
+        self.put(f"teams/{team}/api_key", api_key)
+
+    def set_default_team(self, team: str):
+        if self.get(f"teams/{team}") is None:
+            raise InvalidTeam()
+        self.put("global/default_team", team)
+
+    def set_global(
+        self, api_endpoint: str, base_url: str, directory: str, default_team: Optional[str] = None
+    ):
+        self.put("global/api_endpoint", api_endpoint)
+        self.put("global/base_url", base_url)
+        self.put("global/datasets_dir", directory)
+        if default_team:
+            self.put("global/default_team", default_team)
+
+    def get_team(self, team: str):
+        api_key = self.get(f"teams/{team}/api_key")
+        if api_key is None:
+            raise InvalidTeam()
+        default = (
+            self.get("global/default_team") == team or len(list(self.get("teams").keys())) == 1
+        )
+        return {"slug": team, "api_key": api_key, "default": default}
+
+    def get_default_team(self):
+        default_team = self.get("global/default_team")
+        if default_team:
+            return self.get_team(default_team)
+        teams = list(self.get("teams").keys())
+        if len(teams) > 1:
+            raise InvalidTeam()
```

Comment:
We could make the error more specific. In this case "invalid team" is a bit weird since the user asks for the default one. Maybe we could say something like "no default team set" or along these lines